datamol/align.py

```diff
@@ -0,0 +1,252 @@
+            # with brics scaffold we compute brics bonds, then re-infer and select the biggest
+            # fragment as candidate for core
+            scaffolds = [dm.strip_mol_to_core(x) for x in scaffolds]
+            scaffolds_ids = [dm.to_smiles(x) for x in scaffolds]
+
+        elif partition_method.startswith("anongraph-"):
+            scaffolds = [dm.make_scaffold_generic(s, include_bonds=True) for s in scaffolds]
+            scaffolds_ids = [dm.to_smiles(x) for x in scaffolds]
+
+        elif partition_method.startswith("anon-"):
+            scaffolds = [dm.make_scaffold_generic(s, include_bonds=False) for s in scaffolds]
+            scaffolds_ids = [dm.to_smiles(x) for x in scaffolds]
+
+        for i, s in enumerate(scaffolds_ids):
+            mol_groups[s].append(i)
+            scaffolds[i] = compute_2d_coords(scaffolds[i])
+            scaffold_mols[s] = scaffolds[i]
+
+    elif partition_method == "cluster":
+        # partition is cluster, first compute molecule clusters
+        clusters, mol_clusters = dm.cluster_mols(mols, cutoff=cluster_cutoff, **kwargs)
+
+        # now compute the mcs for each clusters
+        cluster_mcs = [
+            (dm.find_mcs(mol_cluster) if len(mol_cluster) > 1 else dm.to_smiles(mol_cluster[0]))
+            for mol_cluster in mol_clusters
+        ]
+        scaffolds_ids = [cluster_mcs[cluster_id] for cluster_id, _ in enumerate(clusters)]
+
+        for i, s in enumerate(scaffolds_ids):
+            mol_groups[s].extend(clusters[i])
+
+        for x in scaffolds_ids:
+            core = None
+            if x is not None:
+                core = dm.from_smarts(x)
+                core = compute_2d_coords(core)
+            scaffold_mols[x] = core
+
+    else:
+        raise ValueError(f"Unknown partition method: {partition_method}")
+
+    # now we match each molecule to the scaffold and align them
+    # note that the molecule object will be modified in place in the list
+    for cluster_id, (core, mols_ids) in enumerate(mol_groups.items()):
+        core_mol = scaffold_mols[core]
+
+        for mol_id in mols_ids:
+
+            mol = mols[mol_id]
+
+            if core_mol is not None:
+                rdDepictor.GenerateDepictionMatching2DStructure(
+                    mol,
+                    reference=core_mol,
+                    acceptFailure=True,
+                    allowRGroups=True,
+                )
+
+            # Add some props to the mol so the user can retrieve the groups from
+            # it later.
+            props = {}
+            props["dm.auto_align_many.cluster_id"] = cluster_id
+            props["dm.auto_align_many.core"] = core
+            dm.set_mol_props(mol, props)
+
+    # EN: you can discard the mol_groups (or keep it and match the values
+    # to molecular line notation, so you will not have to reocompute the above)
+    # and convert the mols into cxsmiles if you want
+    # return mols, mol_groups
+
+    return mols
```

Comment:
I am trying to keep the outputs consistent and easy to work with, with `df.apply` so i moved the cluster informations to the mol properties.